internal/provider/kubernetes/kubernetes.go

```diff
@@ -41,10 +42,15 @@ func New(cfg *rest.Config, svr *config.Server, resources *message.ProviderResour
 		MetricsBindAddress:     ":8080",
 	}
 
-	if svr.EnvoyGateway.Provider != nil &&
+	// TODO: implment config validatoin on the watch mode config
```

Comment:
There doesn't seem to be any validation in place. Do I need to add it in this PR?